src/gluonts/model/deepstate/issm.py

```diff
@@ -0,0 +1,323 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import List, Tuple
+
+from gluonts.core.component import validated
+from gluonts.distribution.distribution import getF
+from gluonts.model.common import Tensor
+from gluonts.support.util import _broadcast_param
+from gluonts.time_feature.lag import get_granularity
+from gluonts.time_feature import (
+    TimeFeature,
+    MinuteOfHour,
+    HourOfDay,
+    DayOfWeek,
+    WeekOfYear,
+    MonthOfYear,
+)
+
+
+def _make_block_diagonal(blocks: List[Tensor]) -> Tensor:
+    assert (
+        len(blocks) > 0
+    ), "You need at least one tensor to make a block-diagonal tensor"
+
+    if len(blocks) == 1:
+        return blocks[0]
+
+    F = getF(blocks[0])
+
+    # transition coefficient is block diagonal!
+    block_diagonal = _make_2_block_diagonal(F, blocks[0], blocks[1])
+    for i in range(2, len(blocks)):
+        block_diagonal = _make_2_block_diagonal(
+            F=F, left=block_diagonal, right=blocks[i]
+        )
+
+    return block_diagonal
+
+
+def _make_2_block_diagonal(F, left: Tensor, right: Tensor) -> Tensor:
+    """
+    Creates a block diagonal matrix of shape (batch_size, m+n, m+n) where m and n are the sizes of
+            the axis 1 of left and right respectively.
+            (batch_size, seq_length, latent_dim, latent_dim)
+    :param left: shape (batch_size, seq_length, m, m)
+    :param right: shape (batch_size, seq_length, n, n)
+    :return: shape (batch_size, seq_length, m+n, m+n)
+    """
+
+    # shape (batch_size, seq_length, m, n)
+    zeros_off_diag = F.broadcast_add(
+        left.slice_axis(
+            axis=-1, begin=0, end=1
+        ).zeros_like(),  # shape (batch_size, seq_length, m, 1)
+        right.slice_axis(
+            axis=-2, begin=0, end=1
+        ).zeros_like(),  # shape (batch_size, seq_length, 1, n)
+    )
+
+    # shape (batch_size, n, m)
+    zeros_off_diag_tr = zeros_off_diag.swapaxes(2, 3)
+
+    # block diagonal: shape (batch_size, seq_length, m+n, m+n)
+    _block_diagonal = F.concat(
+        F.concat(left, zeros_off_diag, dim=3),
+        F.concat(zeros_off_diag_tr, right, dim=3),
+        dim=2,
+    )
+
+    return _block_diagonal
+
+
+class ISSM:
```

Comment:
would be nice to have some documentation here describing what the class does.

Reply:
Added, thanks.